### DISCUSSION TITLE
Is there a way to browse dependencies sources?

### DISCUSSION THREAD

**Discussion by antosha417** [scalameta/nvim-metals#439] [Q&A] [answered]
Hi!
When I do `:lua vim.lsp.buf.definition()` on a method from some library I see that the file with method definition is copied to `./metals/readonly/dependencies/...` And I'm able to go through this file and read the code and the comments.
But there is only one file there. I want to be able do read other files from the same library. Is it possible with metals? 

I know about the command `Find  in Dependency Jars` and it is very helpful when I know what I'm looking for. But sometimes I just want to poke around and find something interesting

**Comment by ckipp01** [comment, marked as answer]
> And I'm able to go through this file and read the code and the comments.
But there is only one file there.

You should also be able to have code navigation working in that file as well, which can lead you to other files. Is this not working?

**Reply by antosha417** [reply]
This works just fine. But I want to see all other files I have in the library. Not only files that I jumped to.

**Reply by ckipp01** [reply]
Not really files, but one way to do this would be to use the `require("metals.tvp").reveal_in_tree()` which will jump you into the tree. Then you'll be able to see at least all the classes and such in that dep.

**Reply by antosha417** [reply]
Cool, thank you. I think this will do
